### PR TITLE
Import Gohai into the agent

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Import Gohai's git history into the main branch of the agent.

### Motivation

The end goal is to define a new API for Gohai living in `pkg/gohai` and archive the old Gohai github repository.

### Additional Notes

This PR is just a combination of the two following PRs:
- https://github.com/DataDog/datadog-agent/pull/16724 : import Gohai's git history as is
- https://github.com/DataDog/datadog-agent/pull/16994 : fix the CI

The imported code is not used yet.
Follow-up PRs will define the new API and use it instead of the current one.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by pgimalac** [comment]
> [CLA assistant check](https://cla.datadoghq.com/DataDog/datadog-agent?pullRequest=17063) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.datadoghq.com/DataDog/datadog-agent?pullRequest=17063) before we can accept your contribution.**17** out of **23** committers have signed the CLA.✅ olivielpeau✅ gmmeyer✅ derekwbrown✅ arbll✅ truthbk✅ julien-lebot✅ zippolyte✅ ian28223✅ djmitche✅ albertvaka✅ mx-psi✅ pgimalac✅ iglendd✅ sgnn7✅ hush-hush✅ KSerrania✅ dependabot[bot]❌ degemer❌ tariq1890❌ toyozaki❌ davidnewman❌ step-security-bot❌ jauderhoYou have signed the CLA already but the status is still pending? Let us [recheck](https://cla.datadoghq.com/check/DataDog/datadog-agent?pullRequest=17063) it.

Note: this was discussed with open-source who told us we can ignore.

File: tasks/libs/copyright.py [hunk 1]
```diff
@@ -54,6 +54,7 @@
     '^// This file includes software developed at CoreOS',
     '^// Copyright 2017 Kinvolk',
     '^// Copyright 2021 The Vitess Authors.',
+    '^// This file is licensed under the MIT License.',
```
Comment (bkabrda) [inline tasks/libs/copyright.py:57]: I'm slightly worried that this is a little generic and could appear in random opensource projects for which we wouldn't enforce the rules by mistake if we were to add this. Would it be possible to find a more targeted way to do this?

Comment (KSerrania) [reply tasks/libs/copyright.py:57]: AFAIK this is only used in the `lint-copyrights` task, which ensures that a copyright header is present in each `.go` file in the `datadog-agent` repository (it doesn't look at the code from other repositories we import, that is something the `lint-licenses` task does). This rule wouldn't change that behavior (what it says is that the MIT license header is a valid copyright header), so is there a specific risk here to adding this rule you're thinking of?

cc @sgnn7 who may have more context on what the `lint-copyrights` task should do in this case.

**Review by Kaderinho — APPROVED** [review]

File: tasks/libs/copyright.py (hunk 1 above)
Comment (bkabrda) [reply tasks/libs/copyright.py:57]: You're right, I thought these would be also used on files from other projects, but they're not... In which case I think we're safe here and I can approve.

**Review by bkabrda — APPROVED** [review]

File: tasks/libs/copyright.py (hunk 1 above)
Comment (sgnn7) [reply tasks/libs/copyright.py:57]: @KSerrania / @bkabrda Just caught up with this comment and it's already merged but yes, this is used for checking if an approved copyright header is in all `.go` files in strictly just this repo so I think we're good.

**Pull request merged**